src/envs/behavior_options.py

```diff
@@ -0,0 +1,1012 @@
+    """Simple utility function to compute the volume of an aabb."""
+    dimension = hi - lo
+    return dimension[0] * dimension[1] * dimension[2]
+
+
+def get_closest_point_on_aabb(xyz: List, lo: np.ndarray,\
+    hi: np.ndarray) -> List[float]:
+    """Get the closest point on an aabb from a particular xyz coordinate."""
+    closest_point_on_aabb = [0.0, 0.0, 0.0]
+    for i in range(3):
+        # if the coordinate is between the min and max of the aabb, then
+        # use that coordinate directly
+        if xyz[i] < hi[i] and xyz[i] > lo[i]:
+            closest_point_on_aabb[i] = xyz[i]
+        else:
+            if abs(xyz[i] - hi[i]) < abs(xyz[i] - lo[i]):
+                closest_point_on_aabb[i] = hi[i]
+            else:
+                closest_point_on_aabb[i] = lo[i]
+
+    return closest_point_on_aabb
+
+
+def reset_and_release_hand(env) -> None:  # type: ignore
+    """Resets the state of the right hand."""
+    env.robots[0].set_position_orientation(env.robots[0].get_position(),
+                                           env.robots[0].get_orientation())
+    for _ in range(50):
+        env.robots[0].parts["right_hand"].set_close_fraction(0)
+        env.robots[0].parts["right_hand"].trigger_fraction = 0
+        p.stepSimulation()
+
+
+def get_delta_low_level_base_action(  # type: ignore
+    env,
+    original_orientation: Tuple,
+    old_xytheta: np.ndarray,
+    new_xytheta: np.ndarray,
+) -> np.ndarray:
+    """Get low-level actions from base movement plan."""
+    ret_action = np.zeros(17)
+
+    robot_z = env.robots[0].get_position()[2]
+
+    # First, get the old and new position and orientation in the world
+    # frame as numpy arrays
+    old_pos = np.array([old_xytheta[0], old_xytheta[1], robot_z])
+    old_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             old_xytheta[2]]))
+    new_pos = np.array([new_xytheta[0], new_xytheta[1], robot_z])
+    new_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             new_xytheta[2]]))
+
+    # Then, simply get the delta position and orientation by multiplying the
+    #  inverse of the old pose by the
+    # new pose
+    inverted_old_pos, inverted_old_orn_quat = p.invertTransform(
+        old_pos, old_orn_quat)
+    delta_pos, delta_orn_quat = p.multiplyTransforms(inverted_old_pos,
+                                                     inverted_old_orn_quat,
+                                                     new_pos, new_orn_quat)
+
+    # Finally, convert the orientation back to euler angles from a quaternion
+    delta_orn = p.getEulerFromQuaternion(delta_orn_quat)
+
+    ret_action[0:3] = np.array([delta_pos[0], delta_pos[1], delta_orn[2]])
+
+    return ret_action
+
+
+#################
+
+
+# Navigate To #
+def navigate_to_param_sampler(  # type: ignore
```

Comment:
same question, in general i'm not sure what's the motivation for all these `type: ignore`s

Reply:
See above response on the type of `env`.